### PR TITLE
Fix cpp11 compiler errors

### PR DESCRIPTION
**Pull request by marvin2k**
things cropping up while testing for c++11

### CODE REVIEW

File: test/test_memory_layout.cc [hunk 1]
```diff
@@ -111,7 +111,7 @@ BOOST_AUTO_TEST_CASE(test_layout_arrays)
         Arrays arrays_v;
         size_t expected[] = {
             MemLayout::FLAG_MEMCPY,
-               reinterpret_cast<uint8_t*>(&arrays_v.a_v_numeric) - reinterpret_cast<uint8_t*>(&arrays_v),
+               static_cast<size_t>(reinterpret_cast<uint8_t*>(&arrays_v.a_v_numeric) - reinterpret_cast<uint8_t*>(&arrays_v)),
```
Comment (doudou) [inline test/test_memory_layout.cc]: better use ptrdiff_t than size_t then, which is also C++98-compliant

**Comment by marvin2k** [comment]
updated the "narrowing" commit to use `ptrdiff_t`

File: test/test_memory_layout.cc [hunk 2]
```diff
@@ -111,7 +111,7 @@ BOOST_AUTO_TEST_CASE(test_layout_arrays)
         Arrays arrays_v;
         size_t expected[] = {
             MemLayout::FLAG_MEMCPY,
-               reinterpret_cast<uint8_t*>(&arrays_v.a_v_numeric) - reinterpret_cast<uint8_t*>(&arrays_v),
+               static_cast<ptrdiff_t>(reinterpret_cast<uint8_t*>(&arrays_v.a_v_numeric) - reinterpret_cast<uint8_t*>(&arrays_v)),
```
Comment (doudou) [inline test/test_memory_layout.cc]: Not what I meant.... Sorry :P AFAIK, taking the difference of two pointers in C++ is supposed to return std::ptrdiff_t, so by changing the type of expected[] you should be able to remove the static_cast completely. No ?

Comment (marvin2k) [inline test/test_memory_layout.cc]: but then all the other initializer of the array need casting into the ´ptrdiff_t` type?

Comment (doudou) [inline test/test_memory_layout.cc]: OK ... I very probably am telling you something stupid. Let me try it first.

Comment (doudou) [inline test/test_memory_layout.cc]: Got your point ... I am so very sorry to have wasted your time like this.

Comment (doudou) [inline test/test_memory_layout.cc]: Especially for a test. Just noticed ... God you should have told me to get a Prozac or something.

Comment (marvin2k) [inline test/test_memory_layout.cc]: Well, now I'm telling you to get a Prozac ;-)

No problem, I enjoy getting proper proofreads: This is an honour, not a burden.

File: test/test_memory_layout.cc (hunk 1 above)
Comment (marvin2k) [inline test/test_memory_layout.cc]: Keeping `size_t` some after [discussion](https://github.com/orocos-toolchain/typelib/pull/46#discussion-diff-26955253)...

File: test/test_memory_layout.cc (hunk 2 above)
Comment (doudou) [inline test/test_memory_layout.cc]: good point ... offsetof would seem to be a good contender. Will be tricky for the non-POD types though... Anyways. I'm merging as-is.

**Pull request merged**